tools/restatectl/src/commands/log/reconfigure.rs

```diff
@@ -0,0 +1,165 @@
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::num::{NonZeroU32, NonZeroU8};
+
+use anyhow::Context;
+use cling::prelude::*;
+use tonic::codec::CompressionEncoding;
+use tonic::transport::Channel;
+
+use restate_admin::cluster_controller::protobuf::cluster_ctrl_svc_client::ClusterCtrlSvcClient;
+use restate_admin::cluster_controller::protobuf::{
+    ListLogsRequest, ProviderKind as ProtoProviderKind, SealAndExtendChainRequest,
+};
+use restate_types::logs::metadata::{Logs, ProviderKind};
+use restate_types::logs::LogId;
+use restate_types::protobuf::common::Version;
+use restate_types::replicated_loglet::{NodeSet, ReplicatedLogletParams, ReplicationProperty};
+use restate_types::storage::StorageCodec;
+use restate_types::{GenerationalNodeId, PlainNodeId};
+
+use crate::app::ConnectionInfo;
+use crate::util::grpc_connect;
+
+#[derive(Run, Parser, Collect, Clone, Debug)]
+#[cling(run = "reconfigure")]
+pub struct ReconfigureOpts {
+    /// LogId/Partition to seal and extend
+    #[clap(long)]
+    log_id: u32,
+    /// Option segment index to seal, chosen automatically if not provided.
+    #[clap(long)]
+    segment_index: Option<u32>,
+    /// Min metadata version
+    #[clap(long, default_value = "1")]
+    min_version: NonZeroU32,
+    /// Provider kind.
+    #[clap(long, default_value = "replicated")]
+    provider: ProviderKind,
+
+    /// Replication property
+    /// If not provided, it will be copied from the last segment if (and only if) it was
+    /// of replicated type.
+    #[clap(long)]
+    replication_factor: Option<NonZeroU8>,
+    /// A comma-separated list of the nodes in the nodeset. e.g. N1,N2,N4 or 1,2,3
+    /// If not provided, it will be copied from the last segment if (and only if) it was
+    /// of replicated type.
+    #[clap(long, value_delimiter=',', num_args = 0..)]
+    nodeset: Vec<PlainNodeId>,
+    /// The generational node id of the sequencer node, e.g. N1:1
+    /// If not provided, it will be copied from the last segment if (and only if) it was
+    /// of replicated type.
+    #[clap(long)]
+    sequencer: Option<GenerationalNodeId>,
+}
+
+async fn reconfigure(connection: &ConnectionInfo, opts: &ReconfigureOpts) -> anyhow::Result<()> {
+    let channel = grpc_connect(connection.cluster_controller.clone())
+        .await
+        .with_context(|| {
+            format!(
+                "cannot connect to cluster controller at {}",
+                connection.cluster_controller
+            )
+        })?;
+
+    let mut client =
+        ClusterCtrlSvcClient::new(channel).accept_compressed(CompressionEncoding::Gzip);
+
+    let (provider, params) = match opts.provider {
+        ProviderKind::Local => (ProtoProviderKind::Local, rand::random::<u64>().to_string()),
+        #[cfg(any(test, feature = "memory-loglet"))]
+        ProviderKind::InMemory => (ProtoProviderKind::InMemory, String::default()),
```

Comment:
Oversight? `InMemory` loglets also should use u64 random loglet ids in params (like Local loglet) as a string.